codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDecoder.java

```diff
@@ -121,13 +122,15 @@
      */
     public void decode(int streamId, ByteBuf in, Http2Headers headers) throws Http2Exception {
         int index = 0;
-        long headersLength = 0;
         int nameLength = 0;
         int valueLength = 0;
         byte state = READ_HEADER_REPRESENTATION;
         boolean huffmanEncoded = false;
         CharSequence name = null;
         IndexType indexType = IndexType.NONE;
+
+        HeadersDecodingState headersDecodingState = new HeadersDecodingState(streamId, headers,
```

Comment:
can we avoid addition object allocation in the headers decode operation? This maybe a frequently called operation and if we can avoid intermediate allocations that would be preferred.

Reply:
+1... would be nice if we could just store these directly